Mathlib/Probability/Independence/Conditional.lean

```diff
@@ -737,6 +739,34 @@ lemma condIndepFun_self_right {mβ : MeasurableSpace β} {mβ' : MeasurableSpace
     CondIndepFun (mβ'.comap Z) hZ.comap_le X Z μ :=
   condIndepFun_of_measurable_right hX (comap_measurable Z)
 
+theorem condIndepFun_iff_compProd_map_prod_eq_compProd_prod_map_map
```

Comment:
add docstrings for this one and the next ones explaining the statements (and not only paraphrasing them)? They are a little too involved to be parsed comfortably.